.github/workflows/ci.yml

```diff
@@ -183,3 +183,49 @@ jobs:
       run: make clippy-full
     - name: Lint (release)
       run: make clippy-release
+  minimal_versions:
+    name: Minimal versions
+    strategy:
+      matrix:
+        build: [ linux, windows, mac, minimal, default ]
+        include:
+          - build: linux
+            os: ubuntu-latest
+            rust: "stable"
+            features: "full"
+          - build: windows
+            os: windows-latest
+            rust: "stable"
+            features: "full"
+          - build: mac
+            os: macos-latest
+            rust: "stable"
+            features: "full"
+          - build: minimal
+            os: ubuntu-latest
+            rust: "stable"
+            features: "minimal"
+          - build: default
+            os: ubuntu-latest
+            rust: "stable"
+            features: "default"
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: 1.54.0  # MSRV
+          override: true
+      - name: Downgrade crates to minimal supported versions
+        run: cargo +nightly update -Z minimal-versions
+      - uses: Swatinem/rust-cache@v1
+      - name: Build
+        run: make build-${{matrix.features}}
+      - name: Test
+        run: make test-${{matrix.features}}
```

Comment:
Like with escargot, I think I'd rather these be `check` than `test`

Like with MSRV, I'm assuming 99% of the time, a minimal version issue arises from an incompatibility with the API rather than a behavior change.

You have more experience with this though, so let me know if you've been seeing significant issues with relying on new behavior with the same APIs